File: selfstudy/golang/players_reverse.go
/*
task 268:
Players - Reverse
You need to reverse the order of player heights in cm
*/

package main

import (
	"fmt"
	"reflect"
)

func Reverse_orderPlayers(heights []int) []int {
	res := make([]int, len(heights))
	for i, v := range heights {
		res[len(heights)-1-i] = v
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Reverse_orderPlayers([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Reverse_orderPlayers([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Reverse_orderPlayers([]int{1, 2, 3})},
		{[]int{100, 200}, Reverse_orderPlayers([]int{100, 200})},
		{[]int{}, Reverse_orderPlayers([]int{})},
	}
	
	for idx, t := range tests {
		res := Reverse_orderPlayers(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works

File: selfstudy/js/players_reverse.js
/*
task 68:
Players - Reverse
You need to reverse the order of player heights in cm
*/

function reverse_order_players(heights) {
  return [...heights].reverse();
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: reverse_order_players([10, 20, 30]) },
  { input: [50, 60, 70], expected: reverse_order_players([50, 60, 70]) },
  { input: [1, 2, 3], expected: reverse_order_players([1, 2, 3]) },
  { input: [100, 200], expected: reverse_order_players([100, 200]) },
  { input: [], expected: reverse_order_players([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('reverse_order_players', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(reverse_order_players(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, reverse_order_players(t.input) === t.expected || JSON.stringify(reverse_order_players(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think
